api/v1alpha1/dnspolicy_types.go

```diff
@@ -133,7 +133,7 @@ type DNSPolicyStatus struct {
 	HealthCheck *HealthCheckStatus `json:"healthCheck,omitempty"`
 
 	// +optional
-	ProbeConditions map[string][]metav1.Condition `json:"probeConditions,omitempty"`
+	RecordConditions map[string][]metav1.Condition `json:"recordConditions,omitempty"`
```

Comment:
Makes more sense than `probeConditions`. I do wonder how we intend to this for other policies though, i had expected a more policy generic way of reporting status about controller sub resources.